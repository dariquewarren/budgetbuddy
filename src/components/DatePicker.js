import React from 'react';
import { DatePicker } from 'react-rainbow-components';

const initialState = {
    date: undefined,
};
const containerStyles = {
    maxWidth: 400,
};

<div
    className="rainbow-align-content_center rainbow-m-vertical_large rainbow-p-horizontal_small rainbow-m_auto"
    style={containerStyles}
>
    <DatePicker
        id="datePicker-17"
        label="DatePicker Label"
        placeholder="Select date"
        variant="double"
        value={state.date}
        onChange={date => setState({ date })}
    />
</div>