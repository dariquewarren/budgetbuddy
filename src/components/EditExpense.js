import React from 'react'
const EditExpense = (props) =>{

    console.log(props)
 return   (
        <div>
        editing expense with id of {props.match.params.id}
        </div>
        )
}



export default EditExpense