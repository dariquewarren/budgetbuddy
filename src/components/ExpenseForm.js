import React from "react";
import moment from "moment";
import Datetime from 'react-datetime';


var yesterday = moment().subtract( 1, 'day' );
var valid = function( current ){
    return current.isAfter( yesterday );
};
export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString(): "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      error: "",
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = (date) => {
    this.setState(() => ({ createdAt: date }));
  };

  

  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      // set error state to 'please provide descroption and amoint
      this.setState(() => ({ error: "Please provide description and Amount" }));
    } else {
      // clear the error
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt,
        note: this.state.note,
      });
    }
  };

  render() {
    return (
      <div>
        ExpenseForm
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />

          
          <Datetime
          dateFormat="MM-DD-YYYY" 
          timeFormat={false}
          value={this.createdAt}
          initialValue={'created at date'}
          input={true}
          onChange={this.onDateChange}
          isValidDate={valid}
      /> 
          <br></br>
          <textarea
            placeholder="add note for your expense (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          />
          <button>add expense</button>
        </form>
      </div>
    );
  }
}

