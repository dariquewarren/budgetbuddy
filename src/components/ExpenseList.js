import React from 'react'
import { connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
const ExpenseList = (props)=>(
    <div>
    <h1>Expense List</h1>
  
    {props.expenses.length}
    {props.expenses.map((expense)=>{
        return <ExpenseListItem key={expense.id} {...expense}/>
    })}
    
    </div>
)



 const mapStateToProps =connect((state)=>{
    return {
        expenses: state.expenses,
        filters: state.filters
    }
})(ExpenseList)

export default mapStateToProps
