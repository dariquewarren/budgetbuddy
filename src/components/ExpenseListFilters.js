import React from "react";
import { connect } from "react-redux";
import Datetime from 'react-datetime';

import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate,
} from "../actions/filters";


const detailStyles = {
  width: '18rem'
}


class ExpenseListFilters extends React.Component {
  
constructor(props){
  super(props);
  this.state = {
    startDate: 'set start date',
    endDate: 'set end date'
  };
}
  onDatesChange = (value) => {
    
    this.props.dispatch(setStartDate(value));
    this.props.dispatch(setEndDate(value));
  };

  onSetStartDate = (value)=>{
    this.props.dispatch(setStartDate(value))
  }

  onSetEndDate = (value)=>{
    this.props.dispatch(setEndDate(value));

  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));

            console.log(e.target.value);
          }}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === "amount") {
              this.props.dispatch(sortByAmount());
            } else if (e.target.value === "date") {
              this.props.dispatch(sortByDate());
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
      

<details styles={detailStyles}>
<summary>Filter by date range</summary>
<details>
<summary>set a start date </summary>
<Datetime
        dateFormat="MM-DD-YYYY" 
        timeFormat={false}
        value={this.state.startDate}
        initialValue={this.state.startDate}
        input={false}
        onChange={this.onSetStartDate}

    /> 
</details>

<details>
<summary>set an end date</summary>
<Datetime
dateFormat="MM-DD-YYYY" 
timeFormat={false}
value={this.createdAt}
initialValue={this.state.endDate}
input={false}
onChange={this.onSetEndDate}

/>
</details>

</details>


   

       
      </div>
    );
  }
}




// setup value and onchane for select
const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
