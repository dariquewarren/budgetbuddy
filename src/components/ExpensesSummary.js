import React from 'react'
import {connect} from 'react-redux'
import expensesTotal from '../selectors/expenses-total'



const ExpensesSummary = (props)=>{
console.log('props',props.expenses)
if (props.expenses === undefined){
    return <p>no expenses</p>
} else{
   return <h3>
    {props.expenses.length !== 1 ? <p>Viewing {props.expenses.length} expenses totalling ${expensesTotal(props.expenses)} </p> : <p>Viewing {props.expenses.length} expense totalling ${expensesTotal(props.expenses)}</p> }</h3>
    
}
    console.log(expensesTotal(props.expenses))
   
} 
const mapStateToProps = (state)=>{
    return {
        expenses: state.expenses
    }
}
export default connect(mapStateToProps)(ExpensesSummary)

/*

create expense summary component
-----------
renderd by expensedashboardpage
test with 2 snapshot tests
connected to the store for:
-expense count (how many visible expenses) 
-expensesTotal(what's the total of the visible expenses?)

EXAMPLES:

Viewing one expense totalling 95.35

Viewing two expenses totalling 95.35


commit and deploy
get the feature live!
*/

