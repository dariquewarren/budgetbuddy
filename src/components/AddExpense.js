import React from 'react'
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'
import ExpenseForm from './ExpenseForm'
import {connect} from 'react-redux'
import { addExpense } from '../actions/expenses'

const AddExpense = (props)=>(
    <div>
    <h1> Expense form</h1>
    <ExpenseForm
    onSubmit={(expense)=>{
        props.dispatch(addExpense(expense))
        props.history.push('/')
    }}
    />

    </div>
    )

export default connect()(AddExpense)

