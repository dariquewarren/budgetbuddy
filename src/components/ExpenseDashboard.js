import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboard = ()=>(
    <div>
    this is from my dash component
    <ExpenseListFilters/>
    <ExpenseList/>

    </div>
    )

export default ExpenseDashboard







// import React from 'react'
// import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'




// export default 