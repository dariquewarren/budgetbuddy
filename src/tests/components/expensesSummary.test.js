import React from 'react'
import { shallow } from "enzyme";
import {ExpensesSummary} from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'


test('should render snapshot of component with 0 expenses added', ()=>{
    const wrapper = shallow(<ExpensesSummary/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render snapshot of component with expenses added', ()=>{
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()

})