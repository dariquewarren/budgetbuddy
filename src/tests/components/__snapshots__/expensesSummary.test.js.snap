// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot of component with 0 expenses added 1`] = `
<p>
  no expenses
</p>
`;

exports[`should render snapshot of component with expenses added 1`] = `
<h3>
  <p>
    Viewing 
    3
     expenses totalling $
    114195
     
  </p>
</h3>
`;
